src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -10,13 +10,317 @@
+            {
+                if (Unsafe.Add(ref first, start) != trimChar)
+                    break;
+                start++;
+            }
+            return span.Slice(start);
+        }
+
+        /// <summary>
+        /// Removes all trailing occurrences of a specified character.
+        /// </summary>
+        /// <param name="span">The source span from which the character is removed.</param>
+        /// <param name="trimChar">The specified character to look for and remove.</param>
+        public static ReadOnlySpan<char> TrimEnd(this ReadOnlySpan<char> span, char trimChar)
+        {
+            if (span.Length == 0)
+                return default;
+
+            ref char first = ref MemoryMarshal.GetReference(span);
+            int end = span.Length - 1;
+#if !netstandard11
+            if (Vector.IsHardwareAccelerated && span.Length >= 2 * Vector<ushort>.Count)
+            {
+                var mask = new Vector<ushort>(trimChar);
+                do
+                {
+                    end -= Vector<ushort>.Count;
+                    if (Unsafe.ReadUnaligned<Vector<ushort>>(ref Unsafe.As<char, byte>(ref Unsafe.Add(ref first, end))) != mask)
+                    {
+                        end += Vector<ushort>.Count;
+                        break;
+                    }
+                } while (end > Vector<ushort>.Count);
+            }
+#endif
+            while (end >= 0)
+            {
+                if (Unsafe.Add(ref first, end) != trimChar)
+                    break;
+                end--;
+            }
+            return span.Slice(0, end + 1);
+        }
+
+        /// <summary>
+        /// Removes all leading and trailing occurrences of a set of characters specified 
+        /// in a readonly span from the span.
+        /// </summary>
+        /// <param name="span">The source span from which the characters are removed.</param>
+        /// <param name="trimChars">The span which contains the set of characters to remove.</param>
+        public static ReadOnlySpan<char> Trim(this ReadOnlySpan<char> span, ReadOnlySpan<char> trimChars)
+        {
+            if (span.Length == 0)
+                return default;
+            if (trimChars.Length == 0)
+                return span;
+
+            return span.TrimStart(trimChars).TrimEnd(trimChars);
+        }
+
+        /// <summary>
+        /// Removes all leading occurrences of a set of characters specified 
+        /// in a readonly span from the span.
+        /// </summary>
+        /// <param name="span">The source span from which the characters are removed.</param>
+        /// <param name="trimChars">The span which contains the set of characters to remove.</param>
+        public static ReadOnlySpan<char> TrimStart(this ReadOnlySpan<char> span, ReadOnlySpan<char> trimChars)
+        {
+            if (span.Length == 0)
+                return default;
+            if (trimChars.Length == 0)
+                return span;
+
```

Comment:
is it worth to check if trimChars.Length > span.Lenth? I don't think this is a common case though.

Reply:
What should we do differently if trimChars.Length > span.Length?

Reply:
if trimChars.Length > span.Length means there is nothing you need to do and just return the original span because for sure you'll not find anything to trim. this is true for ordinal operation. But as I mentioned this is not a common case so I am not sure if it is worth adding such check

Reply:
> there is nothing you need to do and just return the original span

That is not true. We are looking to trim any of the trimChars, not all of them.

```C#
string temp = "abc";
char[] trimChars = { 'u', 'v', 'w', 'x', 'y', 'z', 'a'};
Console.WriteLine(temp.Trim(trimChars)); // Outputs "bc"
```